Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -682,105 +699,83 @@ namespace AzFramework
     {
         while (!m_stopRequested)
         {
-            if (m_networkImpl->m_gridMate)
+            AZ_PROFILE_SCOPE(AzFramework, "TargetManager::Tick");
+
+            // Send
+            size_t maxMsgsToSend = m_outbox.size();
+            const TargetInfo ti = GetMyTargetInfo();
+            for (size_t iSend = 0; iSend < maxMsgsToSend; ++iSend)
             {
-                AZ_PROFILE_SCOPE(AzFramework, "TargetManager::Tick");
-                if (!m_networkImpl->m_session && !m_networkImpl->m_gridSearch)
-                {
-                    if (AZStd::chrono::system_clock::now() > m_reconnectionTime)
-                    {
-                        JoinNeighborhood();
-                    }
-                }
+                // Send outbound data
+
+                // Lock outbox to prevent a read/write race
+                m_outboxMutex.lock();
 
+                uint8_t* outBuffer = (uint8_t*)m_outbox.front().second.data();
```

Comment:
Avoid c casts, use static cast at least. Do we need any other safety checks here as this code assume a several pieces of data are valid?